lib/AST/ASTScopeSourceRange.cpp

```diff
@@ -40,6 +40,13 @@ static SourceLoc getLocAfterExtendedNominal(const ExtensionDecl *);
 
 void ASTScopeImpl::checkSourceRangeBeforeAddingChild(ASTScopeImpl *child,
                                                      const ASTContext &ctx) const {
+  // Ignore debugger bindings - they're a special mix of user code and implicit
+  // wrapper code that is too difficult to check for consistency.
+  if (auto d = getDeclIfAny().getPtrOrNull())
```

Comment:
We actually need source ranges to be in order for lookups to work correctly, though. Can you ignore the lldb-written part of the binding? It shouldn't have source locations anyway.

Reply:
I believe that’s what this does. When we expand the scopes for the binding’s initializer we’ll still run these checks.